File: two_pointers/3350.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxIncreasingSubarrays = function (nums) {
    const n = nums.length
    let currRun = 1, prevRun = 0, K = 0
    for (let i = 1; i < n; i++) {
        if (nums[i] > nums[i - 1]) {
            currRun++
        }
        else {
            prevRun = currRun
            currRun = 1
        }
        K = Math.max(K, Math.floor(currRun / 2))
        K = Math.max(K, Math.min(prevRun, currRun))
    }
    return K
};

File: two_pointers/3350.cpp
// Tc = O(n)
// Sc = O(1)

class Solution {
public:
    int maxIncreasingSubarrays(vector<int>& nums) {
        int n = nums.size();
        int prevRun = 0;
        int currRun = 1;
        int K = 0;
        for (int i = 1; i < n; i++) {
            if (nums[i] > nums[i - 1]) {
                currRun++;
            } else {
                prevRun = currRun;
                currRun = 1;
            }
            K = max(K, currRun / 2);
            K = max(K, min(prevRun, currRun));
        }
        return K;
    }
};